### PR TITLE
Add an issue template for API proposals

### PR DESCRIPTION
**Pull request by eiriktsarpalis**
Proposals on improving format and wording welcome.

### CODE REVIEW

**Comment by eiriktsarpalis** [comment]
FWIW I don't have sufficient permissions in the repo to use the prescribed [template creation tool](https://help.github.com/en/github/building-a-strong-community/configuring-issue-templates-for-your-repository). The best way around this is to run the tool from the settings in your local fork and take the generated template from there.

File: .github/ISSUE_TEMPLATE/api-proposal.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+about: Propose a change to the public API surface.
+title: "[API Proposal] Your title goes here"
+labels: api-suggestion
+assignees: ''
+
+---
+
+# Background and Motivation
+
+Please provide a thorough explanation of what might necessitate a change to the current APIs.
```
Comment (stephentoub) [inline .github/ISSUE_TEMPLATE/api-proposal.md:12]:
```suggestion
Please provide a thorough explanation of what necessitates the additional API.
```

File: .github/ISSUE_TEMPLATE/api-proposal.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+# Proposed API
+
+Please provide the precise public API signature diff that you are proposing. If modifying existing API signatures this should be expressed using diff blocks, for example
```
Comment (stephentoub) [inline .github/ISSUE_TEMPLATE/api-proposal.md:16]:
```suggestion
Please provide the precise public API signature diff that you are proposing. If augmenting existing API signatures this should be expressed using diff blocks, for example
```

File: .github/ISSUE_TEMPLATE/api-proposal.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+namespace System.Collections.Generic
+{
+-    public class HashSet<T> : ICollection<T>, ISet<T> {
++    public class HashSet<T> : ICollection<T>, ISet<T>, IReadOnlySet<T> {
+     }
+```
+
+# Usage Examples
+
+Please provide code examples that highlight how the proposed API changes are meant to be consumed.
```
Comment (stephentoub) [inline .github/ISSUE_TEMPLATE/api-proposal.md:27]:
```suggestion
Please provide code examples that highlight how the proposed API additions are meant to be consumed.
```

**Comment by eiriktsarpalis** [comment]
Closing in favour of #36431

**Pull request closed without merging**